sdk/servicebus/service-bus/src/core/managementClient.ts

```diff
@@ -301,29 +301,34 @@ export class ManagementClient extends LinkEntity<RequestResponseLink> {
     internalLogger: ServiceBusLogger,
     sendRequestOptions: SendManagementRequestOptions = {}
   ): Promise<RheaMessage> {
+    if (request.message_id == undefined) {
+      request.message_id = generate_uuid();
+    }
     const retryTimeoutInMs =
       sendRequestOptions.timeoutInMs ?? Constants.defaultOperationTimeoutInMs;
     const initOperationStartTime = Date.now();
-    const actionAfterTimeout = () => {
+    const actionAfterTimeout = (reject: (reason?: any) => void) => {
       const desc: string = `The request with message_id "${request.message_id}" timed out. Please try again later.`;
       const e: Error = {
         name: "OperationTimeoutError",
         message: desc
       };
 
-      throw e;
+      reject(e);
     };
 
-    const waitTimer = setTimeout(actionAfterTimeout, retryTimeoutInMs);
-
+    let waitTimer: NodeJS.Timer;
+    const operationTimeout = await new Promise<void>((_, reject) => {
```

Comment:
I don't think you should be awaiting this promise here. Doesn't that cause you to wait for the timeout to be hit before moving onto the next step?

Instead you can just pass the promise to Promise.race below. When you await the Promise.race, it will be fulfilled by whichever promise is fulfilled first.
```suggestion
    const operationTimeout = new Promise<void>((_, reject) => {
```

Reply:
Sorry, that was a leftover, that's what I meant to do.